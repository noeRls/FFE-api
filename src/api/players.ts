import { ChessPlayer, LiscenceType } from '../types/types';
import { getAttributeValue, loadPage, parseElo, text } from './utils';
import { selectOne, selectAll } from 'css-select'

export const fetchPlayer = async (firstname: string, surname: string): Promise<ChessPlayer | undefined> => {
    const document = await loadPage(`/ListeJoueurs.aspx?Action=FFE&JrNom=${surname}&JrPrenom=${firstname}`);
    const row = selectOne('tr[class="liste_clair"]', document);
    if (!row) {
        return undefined;
    }
    const rowItems = selectAll('td', row);
    getAttributeValue(row, 'tr');
    return ({
        firstname,
        surname,
        numberFFE: text(rowItems[0]),
        liscence: text(rowItems[2]) as LiscenceType,
        elo: {
            elo: parseElo(text(rowItems[4])),
            rapid: parseElo(text(rowItems[5])),
            blitz: parseElo(text(rowItems[6])),
        },
        category: text(rowItems[7]),
        club: text(rowItems[9]),
    });
}