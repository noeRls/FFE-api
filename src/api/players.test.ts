import { fetchPlayer } from './players';

it('fetchPlayer', async () => {
    const player = await fetchPlayer("Fabrice", "Moracchini");
    expect(player.numberFFE).toMatchSnapshot();
    expect(player.club.length).toBeGreaterThan(0);
    expect(player.liscence).toMatchSnapshot();
    expect(player.category).toMatchSnapshot();
    expect(player.elo.blitz).toBeGreaterThan(2000);
    expect(player.elo.elo).toBeGreaterThan(2000);
    expect(player.elo.rapid).toBeGreaterThan(2000);
})