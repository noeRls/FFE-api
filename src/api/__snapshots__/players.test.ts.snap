// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPlayer 1`] = `"A07742"`;

exports[`fetchPlayer 2`] = `"A"`;

exports[`fetchPlayer 3`] = `"SepM"`;
